### PR TITLE
Fix auth revoke err not being invoked in async error handler

### PR DESCRIPTION
**Pull request by wallyqs**

### CODE REVIEW

File: nats.go [hunk 1]
```diff
@@ -2757,6 +2758,9 @@ func checkAuthError(e string) error {
 	if strings.HasPrefix(e, AUTHENTICATION_EXPIRED_ERR) {
 		return ErrAuthExpired
 	}
+	if strings.HasPrefix(e, `User Authentication Revoked`) {
```
Comment (kozlovic) [inline nats.go:2761]: Make this a constant and all lower case. We convert the content of error string to lower (see line 2773)

**Review by kozlovic — CHANGES_REQUESTED** [review]

File: nats.go (hunk 1 above)
Comment (kozlovic) [reply nats.go:2761]: (add after `AUTHENTICATION_EXPIRED_ERR` const)

**Comment by matthiashanel** [comment]
@kozlovic just for completeness. this will change how that error is surfaced. (for the better, but it's a change)
without this we'd invoke DisconnectErrHandler without error. The revoked error then could be found in nc.LastError()
```go
nats.DisconnectErrHandler(func(nc *nats.Conn, err error) {
	log.Printf("Disconnected due to: %s, will attempt reconnect", err, nc.LastError())
}),
```

With this change the error handler is invoked separately
```
19:46:00 Unexpected NATS error: nats: authentication revoked <--- ErrorHandler
19:46:00 Disconnected due to: EOF, will attempt reconnect <---- DisconnectErrHandler
19:46:02 Unexpected NATS error: nats: authorization violation <--- ErrorHandler
19:46:04 Unexpected NATS error: nats: authorization violation <--- ErrorHandler
```

**Review by matthiashanel — COMMENTED** [review]
I concur with Ivan.

**Review by kozlovic — APPROVED** [review]
LGTM, and good to have added a test!

**Pull request merged**